File: 2616-maximal-score-after-applying-k-operations/2616-maximal-score-after-applying-k-operations.cs
public class Solution {
    public long MaxKelements(int[] nums, int k) {
        PriorityQueue<int, int> maxHeap = new PriorityQueue<int, int>();
        foreach(int num in nums)
            maxHeap.Enqueue(num , -num);

        long score = 0 ;
        for(int i = 0 ; i < k ; i++) {
            int largest = maxHeap.Dequeue();
            score += largest ;
            int reduced = (int)Math.Ceiling((double)largest / 3);
            maxHeap.Enqueue(reduced , -reduced);
        }
        return score;
    }
}

File: 2616-maximal-score-after-applying-k-operations/2616-maximal-score-after-applying-k-operations.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var maxKelements = function (nums, k) {
    const pq = new MaxPriorityQueue({ compare: (a, b) => b - a })
    for (const e of nums) {
        pq.enqueue(e)
    }

    let s = 0
    while (k--) {
        const e = pq.dequeue()
        s += e
        pq.enqueue(Math.ceil(e / 3))
    }
    return s
};